Add new variant search capabilities

susannasiebert opened the pull request:
This PR adds variant search by:
- name
- allele registry CAID
- non-build37 coordinates (right now only GRCh37 and GRCh38 are supported)

TODO:
- ~Add tests~
- Add support for additional reference build versions

ahwagner commented on civicpy/civic.py:
```diff
@@ -995,6 +996,27 @@ def get_all_variant_groups(allow_cached=True):
     variant_groups = _get_elements_by_ids('variant_group', allow_cached=allow_cached, get_all=True)
     return variant_groups
 
+def search_variants_by_allele_registry_id(caid):
+    """
+    Search the cache for variants matching the queried Allele Registry ID (CAID)
+
+    :param String caid: Allele Registry ID to query
+
+    :return: Returns a list of variant hashes matching the Allele Registry ID
+    """
+    variants = get_all_variants()
+    return [v for v in variants if v.allele_registry_id == caid]
+
+def search_variants_by_name(name):
+    """
+    Search the cache for variants matching the queried name
+
+    :param String name: Variant name to query
+
+    :return: Returns a list of variant hashes matching the name
+    """
+    variants = get_all_variants()
+    return [v for v in variants if v.name == name]
```
I think we should use this opportunity to create a generalized `search_variants_by_attribute` (or, even better, `search_class_by_attribute`) that mirrors this pattern, with appropriate params passed for the explicit routines. Will be less maintenance down the road.

susannasiebert replied:
This has been implemented

susannasiebert commented:
@ahwagner I added the documentation for the two new search functions. I also added a catch guard to the bulk coordinate search since it doesn't currently support search by non-GRCh37 builds.
